src/topp/IDFileConverter.cpp

```diff
@@ -566,6 +585,85 @@ class TOPPIDFileConverter :
         XQuestResultXMLFile().load(in, peptide_identifications, protein_identifications);
       }
 
+      else if (in_type == FileTypes::FASTA)
+      {
+        MSExperiment exp;
+        TheoreticalSpectrumGenerator tsg;
+
+        // extract parameters and remove non tsg params
+        Param p = getParam_().copy("fasta_to_mzml:", true);
+        String enzyme = p.getValue("enzyme");
+        Int mc = p.getValue("missed_cleavages");
+        Int min_charge = p.getValue("min_charge");
+        Int max_charge = p.getValue("max_charge");
+        Int pc_charge = p.getValue("precursor_charge");
+        p.remove("enzyme");
+        p.remove("missed_cleavages");
+        p.remove("min_charge");
+        p.remove("max_charge");
+        p.remove("precursor_charge");
+
+        if (min_charge > max_charge)
+        {
+          writeLog_("Error: 'fasta_to_mzml:min_charge' must be smaller than or equal to 'fasta_to_mzml:max_charge'.");
+          printUsage_();
+          return ILLEGAL_PARAMETERS;
+        }
+
+        if (pc_charge != 0 && pc_charge <= max_charge)
+        {
+          writeLog_("Error: 'fasta_to_mzml:precursor_charge' must be bigger than 'fasta_to_mzml:max_charge'.\nSet 'precursor_charge' to '0' to automaticly use 'max_charge' + 1.");
+          printUsage_();
+          return ILLEGAL_PARAMETERS;
+        }
+
+        tsg.setParameters(p);
+
+        ProteaseDigestion digestor;
+        digestor.setEnzyme(enzyme);
+        digestor.setMissedCleavages(mc);
+
+        // loop through fasta input
+        FASTAFile::FASTAEntry entry;
+        FASTAFile f;
+        f.readStart(in);
+        while (f.readNext(entry))
+        {
+          // digest sequence of fasta entry
+          vector<AASequence> digested_peptides;
+          AASequence seq = AASequence::fromString(entry.sequence);
+          digestor.digest(seq, digested_peptides);
+
+          // for each peptide calculate the theoretical spectrum
+          for (const auto& peptide : digested_peptides)
+          {
+            PeakSpectrum spec;
+            tsg.getSpectrum(spec, peptide, min_charge, max_charge, pc_charge);
+            exp.addSpectrum(spec);
+          }
+        }
+        logger.endProgress();
+
+        // handle out type
+        const String out = getStringOption_("out");
+        FileTypes::Type out_type = FileTypes::nameToType(getStringOption_("out_type"));
+        if (out_type != FileTypes::MZML)
```

Comment:
this should be done before doing expensive computations, so the user gets the feedback immediately instead of waiting until the tool is almost done.